_inc/client/state/recommendations/actions.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Internal dependencies
+ */
+import restApi from 'rest-api';
+import {
+	JETPACK_RECOMMENDATIONS_DATA_ADD_SELECTED_RECOMMENDATION,
+	JETPACK_RECOMMENDATIONS_DATA_ADD_SKIPPED_RECOMMENDATION,
+	JETPACK_RECOMMENDATIONS_DATA_FETCH,
+	JETPACK_RECOMMENDATIONS_DATA_FETCH_RECEIVE,
+	JETPACK_RECOMMENDATIONS_DATA_FETCH_FAIL,
+	JETPACK_RECOMMENDATIONS_DATA_UPDATE,
+	JETPACK_RECOMMENDATIONS_DATA_SAVE,
+	JETPACK_RECOMMENDATIONS_DATA_SAVE_SUCCESS,
+	JETPACK_RECOMMENDATIONS_DATA_SAVE_FAIL,
+	JETPACK_RECOMMENDATIONS_STEP_UPDATE,
+	JETPACK_RECOMMENDATIONS_STEP_UPDATE_SUCCESS,
+	JETPACK_RECOMMENDATIONS_STEP_UPDATE_FAIL,
+	JETPACK_RECOMMENDATIONS_UPSELL_FETCH,
+	JETPACK_RECOMMENDATIONS_UPSELL_FETCH_RECEIVE,
+	JETPACK_RECOMMENDATIONS_UPSELL_FETCH_FAIL,
+} from 'state/action-types';
+
+export const fetchRecommendationsData = () => {
+	return dispatch => {
+		dispatch( { type: JETPACK_RECOMMENDATIONS_DATA_FETCH } );
+		return restApi
+			.fetchRecommendationsData()
+			.then( data => {
+				dispatch( { type: JETPACK_RECOMMENDATIONS_DATA_FETCH_RECEIVE, data } );
+			} )
+			.catch( error => {
+				dispatch( { type: JETPACK_RECOMMENDATIONS_DATA_FETCH_FAIL, error } );
+			} );
+	};
+};
+
+export const updateRecommendationsData = data => {
+	return dispatch => {
+		dispatch( { type: JETPACK_RECOMMENDATIONS_DATA_UPDATE, data } );
+	};
+};
+
+const saveRecommendations = ( dispatch, getState ) => {
+	const recommendations = getState().jetpack.recommendations;
+	dispatch( { type: JETPACK_RECOMMENDATIONS_DATA_SAVE } );
+	return restApi
+		.saveRecommendationsData( recommendations.data )
+		.then( () => {
+			dispatch( { type: JETPACK_RECOMMENDATIONS_DATA_SAVE_SUCCESS } );
+		} )
+		.catch( error => {
+			dispatch( { type: JETPACK_RECOMMENDATIONS_DATA_SAVE_FAIL, error } );
+		} );
+};
+
+export const saveRecommendationsData = () => {
```

Comment:
It looks like this function is not used anywhere. Do we still need it?

Reply:
It is used in the UI which is in a different PR, see [here](https://github.com/Automattic/jetpack/pull/17691/files#diff-78e1e217c7c497d7595da78066235a28807c6696269605495bcc84ba6af35149R111) for an example.